### PR TITLE
chore: add unit test for CIDocument filterGroups util

### PR DESCRIPTION
**Pull request by H-Fin**
#### What do these changes do/fix?
Adds unit tests to the CIDocument utils file `filterGroups.ts` for higher test coverage of our code.
**Note:** The tests in `CIDocument.spec.ts` build off of a previous PR #73 

https://zenhub.ibm.com/app/workspaces/discovery-unified-5728bf39d2c20f776e05d0ef/issues/watson-discovery/docviz-squad-issue-tracker/234

#### How do you test/verify these changes?
`yarn test watch:documentData` and `yarn test watch:filterGroups` and see that tests pass.

Run `yarn test:coverage` and view the coverage of the `CIDocument/utils/filterGroups.ts` file to see the results are statements: 100%, branches: 100%, and functions: 100%.

### CODE REVIEW

File: packages/discovery-react-components/src/components/CIDocument/utils/__tests__/filterGroups.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { getBaseFilterGroups, FilterGroupWithFns } from '../filterGroups';
+
+describe('filterGroups', () => {
+  it('handles invalid ducment types correctly', () => {
```
Comment (jhpedemonte) [inline packages/discovery-react-components/src/components/CIDocument/utils/__tests__/filterGroups.spec.ts]: `ducment` -> `document`

Comment (ColbyJohnIBM) [reply packages/discovery-react-components/src/components/CIDocument/utils/__tests__/filterGroups.spec.ts]: It's slang, Javier. Get with the times...

Comment (jhpedemonte) [reply packages/discovery-react-components/src/components/CIDocument/utils/__tests__/filterGroups.spec.ts]: I'm so old 😢

Comment (H-Fin) [reply packages/discovery-react-components/src/components/CIDocument/utils/__tests__/filterGroups.spec.ts]: lol--fixed!

**Pull request merged**